src/app/MainMenu.js

```diff
@@ -0,0 +1,195 @@
+  document.body.insertBefore(menu, placeToAppend);
+  document.body.insertBefore(backdropForSearch, placeToAppend)
+
+  //SEARCH
+  //data to connect with spoonacular
+  const API_KEY = 'a69c65ede3bb4ac3b262c5b425b4f835';
+  const URL = `https://api.spoonacular.com/recipes/complexSearch?query=`
+
+  //elements' of DOM to manipulate
+  const inputForSearch = document.querySelector('form input');
+  const buttonForSearch = document.querySelector('form button'); 
+  const infoForEmptySearch = document.querySelector('.tooltip');
+  const backdrop = document.querySelector('.backdrop');
+  const closeSearchResultBtn = backdrop.firstElementChild.firstElementChild;
+  const resultsSection = backdrop.firstElementChild.lastElementChild;
+
+  //clear input value
+  const clearInput = () => {
+    inputForSearch.value = ''
+  } 
+
+  //clear search info (about empty value)
+  const clearSearchInfo = () => {
+    infoForEmptySearch.classList.remove('active')
+  }
+
+  //create box for one result of search
+  const createResultBox = (data, parentElement) => {
+    const outputBox = document.createElement('article');
+    outputBox.id = data.id;
+    const outputPhoto = document.createElement('img');
+    outputPhoto.src = data.image;
+    const outputTitle = document.createElement('p');
+    outputTitle.innerText = data.title;
+
+    appendChildrenToElement(outputBox, outputPhoto, outputTitle);
+    parentElement.appendChild(outputBox)
+  }
+
+  //function to send request
+  const sendRequest = (value) => {
+    //prepare search text to send request
+    const textToSearch = value.value.trim().replace('', '%20');
+
+    fetch(URL + textToSearch + `&apiKey=${API_KEY}`)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Ups...  Something went wrong!');
+      } 
+      return response.json()
+    })
+    .then(recipes => {
+      if (recipes.results.length === 0) {
+        const noResultsInfo = document.createElement('p');
+        noResultsInfo.innerText = `Sorry, there isn't any result for Your search`;
+        resultsSection.appendChild(noResultsInfo)
+      } else {
+        recipes.results.forEach(recipe => createResultBox(recipe, resultsSection))
+      }
+    })
+    .catch(error => {
+      resultsSection.innerHTML = `<p>${error}</p>`;
+      console.log(error)});
+  }
+
+  inputForSearch.addEventListener('click', e => clearSearchInfo())
+
+
+  //send request and generate output 
+  buttonForSearch.addEventListener( 'click' , e => {
+    e.preventDefault();
+
+    if (inputForSearch.value !== '') {
+    clearSearchInfo();
+
+    //set backdrop visible
+    backdrop.style.opacity = 1;
+    backdrop.style.zIndex = 100;
+
+    //create information about search value in modal
+    const searchInfo = `Results for search: ${inputForSearch.value}`
+    const searchTitle = document.createElement('h2');
+    searchTitle.innerText = searchInfo;
+    resultsSection.appendChild(searchTitle);
+
+    sendRequest(inputForSearch)
```

Comment:
I think you miss try/catch block here, send request will throw error on failure, then this error should be logged, we can think about it next time